### PR TITLE
Add MongoDB replica set example

### PR DESCRIPTION
**Pull request by mfojtik**
TODO:
- [x] Add authentication back and use keyFile for inter-mongo communication
- [ ] Re-create the replicaSet if all Pods died, but there is a persistent storage
- [x] Add more documentation about how this works

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees @jhadvig @danmcp this is a new PR that use one-shot member management Pod that will initialize the replica.

**Comment by mnagy** [comment]
@mfojtik we're using tabs in shell scripts in our images

**Comment by mfojtik** [comment]
@mnagy then we are doing it wrong :-)

**Comment by danmcp** [comment]
@mfojtik +1

**Comment by mfojtik** [comment]
@mnagy configure your vim with `:set expandtab` and then go through all scripts we have and write `:retab` :-)

**Comment by mfojtik** [comment]
I made some refactoring to the names of Pod(s) and containers to make it more clear and to be more descriptive about their roles in the cluster. Also renamed the 'replica_manager.sh' to 'initiate_replica.sh' as the Pod is on-shot and it does not really manage replica Pods.

**Comment by mfojtik** [comment]
@danmcp @jhadvig PTAL (this is now ready for merge)

File: 2.4/contrib/common.sh [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      echo "=> MongoDB service has started"
+      return 0
+    fi
+    sleep $SLEEP_TIME
+  done
+  echo "=> Giving up: Failed to start MongoDB service!"
+  exit 1
+}
+
+# wait_for_mongo_up waits until the mongo server is down
```
Comment (danmcp) [inline 2.4/contrib/common.sh]: wrong name

File: 2.4/contrib/common.sh [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+# endpoints returns list of IP addresses with other instances of MongoDB
+# To get list of endpoints, you need to have headless Service named 'mongodb'.
+# NOTE: This won't work with standalone Docker container.
+function endpoints() {
+  service_name=${MONGODB_SERVICE_NAME:-mongodb}
+  dig ${service_name} A +search +short 2>/dev/null
+}
+
+# no_endpoints returns true if the only endpoint is the current container itself
+# or there are no endpoints registred (running as standalone Docker container)
```
Comment (danmcp) [inline 2.4/contrib/common.sh]: registered

File: 2.4/contrib/replica_supervisor.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+#
+# This script registers the current container into MongoDB replica set and
+# unregister it when the container is terminated.
```
Comment (danmcp) [inline 2.4/contrib/replica_supervisor.sh]: unregisters

File: 2.4/examples/replica/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+**WARNING**: The provided `mongo_controller.json` file contains the base64 encoded keyFile, but it is highly recommended to generate your own private file.
+
+### Pod 'mongo-service'
+
+This Pod serves as an replSet initialization member. Inside the
+'initiate_replica.sh' command is ran, which will wait for all members created by
+the ReplicationController to become available and then creates the initial
+MongoDB replica set. This set is then distributed to all members.
+This Pod is a 'one-shot' Pod, which means that once the replSet is initialized
```
Comment (danmcp) [inline 2.4/examples/replica/README.md]: Is one-shot official terminology?  Why not say run-once?

Comment (mfojtik) [inline 2.4/examples/replica/README.md]: run-once sounds better :)

File: 2.4/examples/replica/README.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+* `MONGODB_ADMIN_PASSWORD`
+
+Optionally you can set `MONGODB_DATABASE` (it defaults to `MONGODB_USERNAME`).
+
+## How to test it?
+
+You can play with the number of replicas in the Replication Controller. Use the
+command above and change the number of replicas as you want. You can tail the
+logs from one of the replica members and see how the new replSet members are
+added and removed automatically. The `docker ps` and `docker exec` are your
```
Comment (danmcp) [inline 2.4/examples/replica/README.md]: I hate to reference docker ps and docker exec here.

**Comment by mfojtik** [comment]
@danmcp fixed, thanks!

**Comment by danmcp** [comment]
[merge]

**Comment by jhadvig** [comment]
@mfojtik what about tests ?

**Pull request merged**